packages/components/src/calendar/date-range.js

```diff
@@ -137,8 +193,14 @@ class DateRange extends Component {
 						onFocus={ () => this.onFocusChange( 'endDate' ) }
 					/>
 				</div>
-				<div className="woocommerce-calendar__react-dates">
+				<div
+					className="woocommerce-calendar__react-dates"
+					ref={ this.nodeRef }
+					onBlur={ partial( this.keepFocusInside, CONTAINER_DIV ) }
```

Comment:
Keyboard navigation doesn't let you tab past the first arrow 😭 . I'm not sure how to distinguish a blur event via keyboard navigation from blur event via mouse. Here is a [suggestion for setting a flag](https://stackoverflow.com/questions/5653332/differentiate-between-focus-event-triggered-by-keyboard-mouse) 🤷‍♀ 

In any case, until an upstream change happens to react-dates, I think this is an acceptable tradeoff considering the inputs are accessible.

Reply:
Keyboard navigation should be fixed with d03971b3.

The one questionable bit is the addition of a `losesFocusTo` prop to `DateRange`. It allows the tabbing to continue past the calendar grid and into the comparison period selector.